pkgs/development/cuda-modules/db/schema.nix

```diff
@@ -0,0 +1,413 @@
+{ lib, config, ... }:
+let
+  inherit (lib) mkOption types;
+  inherit (lib.types)
+    attrsOf
+    bool
+    enum
+    listOf
+    nullOr
+    str
+    submodule
+    ;
+
+  inherit (import ./columnar.nix)
+    Unit
+    SetOfStr
+    mkColumnOption
+    assertSubset
+    ;
+
+  cudb = config;
+in
+{
+  options =
+    let
+      License = submodule (
+        { name, config, ... }:
+        let
+          shortName = name;
+        in
+        {
+          options = {
+            fullName = mkOption {
+              type = str;
+              default = config.shortName;
+              defaultText = "<shortName>";
+              description = "Full (long, descriptive) name of the license";
+            };
+            shortName = mkOption {
+              type = str;
+              default = shortName;
+              defaultText = "<name>";
+              description = ''
+                Short name used to identify the license
+                In `cudaPackages`, licenses from manifests are grouped by `shortName`s
+                and so no two licenses of the same `shortName` may contain conflicting definitions.
+              '';
+            };
+            free = mkOption {
+              type = bool;
+              default = false;
+              description = ''License grants the "four freedoms"'';
+            };
+            redistributable = mkOption {
+              type = bool;
+              default = true;
+              description = ''License may not be "free" but permits redistribution'';
+            };
+            deprecated = mkOption {
+              type = bool;
+              default = false;
+              description = "License definition no longer used in Nixpkgs";
+            };
+            spdxId = mkOption {
+              type = nullOr str;
+              default = null;
+              description = "Optional SPDX Identifier";
+            };
+            url = mkOption {
+              type = nullOr str;
+              default =
+                let
+                  ensureComposable = base: if lib.hasSuffix "/" base then base else "${base}/";
```

Comment:
If the string has a leading slash, pass it through, otherwise append a trailing slash?
Personally I'd rather see something which strips leading and trailing slashes, so we must be explicit in adding them during interpolation.

Reply:
> leading slash

No-no, it's `hasSuffix`, so it just ensures there's a trailing slash